roles/variables/vars/main.yml

```diff
@@ -138,17 +308,8 @@ kafka_broker_properties:
       confluent.support.customer.id: anonymous
       log.dirs: "/var/lib/kafka/data"
       kafka.rest.enable: "{{kafka_broker_rest_proxy_enabled|string|lower}}"
-  controller_mode:
-    enabled: "{{kraft_enabled|bool and inventory_hostname in groups.kafka_controller}}"
-    properties:
-      process.roles: controller
-      controller.quorum.voters: "{% for inventory_hostname in groups.kafka_controller|default([]) %}{% if loop.index > 1%},{% endif %}{{inventory_hostname[-1]|int + 9990}}@{{inventory_hostname}}:9093{%endfor%}"
-      controller.listener.names: "{{kafka_controller_listeners['controller']['name']}}"
-      listener.security.protocol.map: "{% for listener in kafka_controller_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}:{{ listener['value'] | confluent.platform.kafka_protocol_defaults(kafka_controller_ssl_enabled, kafka_controller_sasl_protocol)}}{% endfor %}"
-      listeners: "{{kafka_controller_listeners['controller']['name']}}://:{{kafka_controller_listeners['controller']['port']}}"
-      node.id: "{{ inventory_hostname[-1]|int + 9990 }}"
   broker_on_controller:
-    enabled: "{{kraft_enabled|bool and inventory_hostname in groups.kafka_broker}}"
+    enabled: "{{kraft_enabled|bool}}"
     properties:
       process.roles: broker
       controller.quorum.voters: "{% for inventory_hostname in groups.kafka_controller|default([]) %}{% if loop.index > 1%},{% endif %}{{inventory_hostname[-1]|int + 9990}}@{{inventory_hostname}}:9093{%endfor%}"
```

Comment:
1. 9093 hardcoded.
2. We are deriving this value `controller.quorum.voters` at 2 places(once for each broker and controller). Let's derive it once outside and later use the variable here. So if logic changes in the future, we don't have to change multiple places.

Reply:
now that we have separated kafka_broker_properties and kafka_controller_properties, we would need it at both the places, right? there are more such properties which are repeated cause we need them in both controller and broker properties